docs/ref/middleware.txt

```diff
@@ -495,6 +495,30 @@ Adds the ``user`` attribute, representing the currently-logged-in user, to
 every incoming ``HttpRequest`` object. See :ref:`Authentication in web requests
 <auth-web-requests>`.
 
+.. class:: LoginRequiredMiddleware
+
+.. versionadded:: 5.1
+
+Redirects all requests by non-authenticated users to a login page. For admin 
+views, this will redirect to the admin login. For all other views, this will 
+redirect to :setting:`settings.LOGIN_URL <LOGIN_URL>`. This can be customized 
+by setting ``login_url`` or ``redirect_field_name`` attributes on your view.
+
+Here are examples of how you can apply this in both class based and function 
+based views::
+
+    class MyView(View):
+        login_url = "/login/"
+        redirect_field_name = "redirect_to"
+
+
+    @login_required(redirect_field_name="redirect_to", login_url="/login/")
+    def my_view(request): ...
+
+Views using the :func:`~django.contrib.auth.decorators.login_not_required` 
+decorator or the :class:`~django.contrib.auth.mixins.LoginNotRequiredMixin` 
+mixin are exempt from this requirement.
+
```

Comment:
We potentially want a warning here that the login url (either set on the middleware or the settings) must not require a login itself, otherwise there is an infinite redirect loop
It _should_ be self explanatory but maybe we should call it out in a docs warning 🤔 what do we think?

Reply:
I agree, in addition to middleware or the settings it can also be set on views. But I think a general warning to not require a login on login views is okay?

Reply:
Added something here 👍